.github/workflows/detekt.yaml

```diff
@@ -0,0 +1,24 @@
+name: Checks
+
+on:
+  push:
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up JDK
+        uses: actions/setup-java@v3
+        with:
+          distribution: oracle
+          java-version: '17'
+
+      - name: Run detekt
```

Comment:
In this step, you run the build of the entire project, not just the detekt plugin